### PR TITLE
Feat/add ols enrichment embedding

### LINKED ISSUE #159: Utility functions to enrich the knowledge graph nodes and edges with static features for textual enrichment
# Description
We would like to enrich the knowledge graph with additional ontologies. For example, PrimeKG uses NCBI gene IDs while many biomodels use UniProt IDs to identify protein species. In addition, the mapping from biomodels to BKGs is one to many whereby a single biomodel species includes not just the species name/annotation, but also the compartment which describes the organ/tissue or cellular/subcellular location. Also, species can be modified from their original form e.g., phosphorylation of proteins or splice variants for transcripts, which also need to be accounted for.

The node types we would focus on are the following: 
- Genes: e.g., NCBI Gene, UniProt, GO
- Proteins/Peptides/Antibodies: e.g., UniProt, PO
- Metabolites/Drugs: e.g., ChEBI, PubChem, DrugBank, DrugCentral
- Cells: e.g., CO
- Diseases/Phenotypes: e.g., DO, ICD, Snomed, Mondo, 
- Organ/tissue: e.g., BTO, Uberon, 
- Cellular/subcellular locations: e.g., GO:CL
- Pathways: e.g., GO:MF, GO:BP, Reactome, KeGG, BioCyc
- Relationship: e.g., RO

To map to different ontologies and enrich the knowledge graph with their synonyms, textual descriptions, and optionally cross-references if available, we would like to develop a set of utility functions. An example tool that we created specifically for T2B that can be used as inspiration can be found here: https://github.com/VirtualPatientEngine/AIAgents4Pharma/blob/main/aiagents4pharma/talk2biomodels/tools/get_annotation.py. For implementing the utility functions, we can assume that the user would be able to provide the name of the species in question.

# Checklist
- [ ] General interface for calling external biomedical knowledge bases to lookup a particular species and extract out synonyms, textual descriptions, and other textual features that can be used to enrich a BKG
- [ ] Specialized classes for each of the different node types described above. 
- [ ] Unit tests for testing the behavior of the utility functions
- [ ] Jupyter notebook demonstrating the functionality of the utility functions
- [ ] Updated documentation

# Acceptance criteria
- [ ] Successful PR

### PR DESCRIPTION
**Pull request by gurdeep330**
# For authors

## Description
This PR enriches and embeds the remaining nodes in PrimeKG by incorporating textual descriptions from several curated sources.

Data Sources for Enrichment:
1. CTD (Comparative Toxicogenomics Database) – Mapped CTD IDs to corresponding PubChem IDs to retrieve description.
2. Reactome – Used pathway and reaction descriptions for relevant biological entities.
3. OLS (Ontology Lookup Service) – Queried terms from various biomedical ontologies, including:
    - GO (Gene Ontology)
    - HPO (Human Phenotype Ontology)
    - MONDO (Mondo Disease Ontology)
    - UBERON (Anatomical Ontology)
    - CL (Cell Ontology) (PrimeKG does not have cell ontology nodes, but the same code (`EnrichmentWithOLS`) can be used to enrich nodes with CL in other biomedical graphs)

Additional notes:
1. Updated `EnrichmentWithPubChem` to also return descriptions besides SMILES
2. Standardised pubmed_utils to consider all data sources of PubChem for ID conversions

## Type of change
Please delete options that are not relevant.
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

## How Has This Been Tested?
Please describe the tests you conducted to verify your changes. These may involve creating new test scripts or updating existing ones.
- [X] Added new test(s) in the ```tests``` folder
- [X] Added new function(s) to an existing test(s) (e.g.: ```tests/testX.py```)
- [ ] No new tests added (Please explain the rationale in this case)

## Checklist
- [X] My code follows the style guidelines mentioned in the Code/DevOps guides
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (e.g. MkDocs)
- [X] My changes generate no new warnings
- [X] I have added or updated tests (in the ```tests``` folder) that prove my fix is effective or that my feature works
- [X] New and existing tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

# For reviewers

## Checklist pre-approval
- [ ] Is there enough documentation?  
- [ ] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
- [ ] Does the test(s) successfully test edge/corner cases?
- [ ] Does the PR pass the tests? (if the repository has continuous integration)

## Checklist post-approval
- [ ] Does this PR merge ```develop``` into ```main```? If so, please make sure to add a prefix (feat/fix/chore) and/or a suffix BREAKING CHANGE (if it's a major release) to your commit message.
- [ ] Does this PR close an issue? If so, please make sure to descriptively close this issue when the PR is merged.

## Checklist post-merge
- [ ] When you approve of the PR, merge and close it (Read this [article](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) to know about different merge methods on GitHub)
- [ ] Did this PR merge ```develop``` into ```main``` and is it suppose to run an automated release workflow (if applicable)? If so, please make sure to check under the "Actions" tab to see if the workflow has been initiated, and return later to verify that it has completed successfully.

### CODE REVIEW

**Review by awmulyadi — APPROVED** [review]
Hi, Gurdeep. The PR looks clean, and it is amazing that now we can enrich more nodes by utilizing this new enrichment method 👍 I just want to point out that there are missing documentation files for these new methods under `docs\talk2knowledgegraphs\utils\enrichments`. In addition, such files may need to be included in `aiagents4pharma\talk2knowledgegraphs\utils\enrichments\__init__.py` perhaps to avoid any issues in building the package afterward.

File: aiagents4pharma/talk2knowledgegraphs/utils/enrichments/ols_terms.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            r = requests.get(cfg.base_url,
+                             headers={ "Accept" : "application/json"},
+                             params=params,
+                             timeout=cfg.timeout)
+            response_body = json.loads(r.text)
+            # if the response body is empty
+            if '_embedded' not in response_body:
+                descriptions.append(None)
+                continue
+            description = response_body['_embedded']['terms'][0]['description']
```
[the remark this answers is not in the document]
Comment (gurdeep330) [reply aiagents4pharma/talk2knowledgegraphs/utils/enrichments/ols_terms.py:56]: Thanks for the suggestion. No, labels and synonyms are different fields.
I have updated the code.

**Comment by gurdeep330** [comment]
> Hi, Gurdeep. The PR looks clean, and it is amazing that now we can enrich more nodes by utilizing this new enrichment method 👍 I just want to point out that there are missing documentation files for these new methods under `docs\talk2knowledgegraphs\utils\enrichments`. In addition, such files may need to be included in `aiagents4pharma\talk2knowledgegraphs\utils\enrichments\__init__.py` perhaps to avoid any issues in building the package afterward.

Thanks for pointing it out @awmulyadi 
I have updated the code.
